src/main/scala/cromwell/engine/backend/BackendCall.scala

```diff
@@ -54,21 +54,37 @@ trait ExecutionHandle {
   def result: ExecutionResult
 }
 
+sealed trait ReturnCode {
+  def code: Int
+
+  def asScriptReturnCode = this match {
+    case script: ScriptReturnCode => Option(code)
+    case _ => None
+  }
+
+  def asBackendReturnCode = this match {
+    case script: BackendReturnCode => Option(code)
+    case _ => None
+  }
```

Comment:
Hmm, how about every call returning a `returnCode: Option[ScriptReturnCode]` and a `backendErrorCode: Option[BackendErrorCode]` - with the `EXECUTION_INFO` table only getting an entry if `Option[BackendErrorCode]` is `Some`?